Create workflow for releasing documentation

svij-sc opened the pull request:
**Scope of work done**

<!-- Description of PR goes here -->

Introducing new workflow that will allow us to deploy our docs to github pages.
I have already [merged it to snap-research/GiGL](https://github.com/snap-research/GiGL/actions/runs/14806387587/workflow?pr=21).
You can see the successful results of [deployment](https://github.com/snap-research/GiGL/actions/runs/14806387587/job/41575241673?pr=21) using the rule here: https://snap-research.github.io/GiGL/

<!-- Relevant screenshots go here (optional) -->

Where is the documentation for this feature?: N/A

Did you add automated tests or write a test plan?

***Updated Changelog.md?*** NO

***Ready for code review?:*** NO

kmontemayor2-sc reviewed (commented):
Thanks Shubham! Is it possible to run this prior to merging?

And additionally. where does the doc site get hosted? Should we mention somewhere in the action where it does?

svij-sc commented on .github/workflows/release-documentation.yml:
```diff
@@ -0,0 +1,54 @@
+          path: 'gh_pages_build/html/'
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v4
```
correct.
This is how github expects us to deploy pages now.

svij-sc commented:
@kmontemayor2-sc 

> Thanks Shubham! Is it possible to run this prior to merging?

There is no easy way to do this because of branch protection rules i.e. one would need to be setup specifically for this branch, then I will need to change the trigger to deploy w/ on_push to this branch.
Even if done so, we would not compile anything as we also need changes introduced here:
1. https://github.com/Snapchat/GiGL/pull/25
2. https://github.com/Snapchat/GiGL/pull/26 

I have gone ahead and verified that this works in our Snap-Research repo
> I have already [merged it to snap-research/GiGL](https://github.com/snap-research/GiGL/actions/runs/14806387587/workflow?pr=21).
You can see the successful results of [deployment](https://github.com/snap-research/GiGL/actions/runs/14806387587/job/41575241673?pr=21) using the rule here: https://snap-research.github.io/GiGL/

svij-sc commented:
@kmontemayor2-sc , thanks for the review.

I have tried addressing your comments including:
> And additionally. where does the doc site get hosted? Should we mention somewhere in the action where it does?
